geniza/corpus/admin.py

```diff
@@ -592,3 +594,32 @@ def save_model(self, request, obj, form, change):
         # if there is an error loading the IIIF manifest
         obj.request = request
         super().save_model(request, obj, form, change)
+
+
+admin.site.unregister(Tag)
+
+
+@admin.register(Tag)
+class CustomTagAdmin(TagAdmin):
+    list_display = ("name", "slug", "document_count")
+
+    def get_queryset(self, request):
+        # The annotations we use for document count on the list view
+        # make the search too slow for autocomplete.
+        # Reset to original, unannotated queryset *only* for autocomplete
+        qs = super().get_queryset(request)
+        if request and request.path == "/admin/autocomplete/":
+            # return without annotations
+            return qs
```

Comment:
Did you test if taggit uses admin querysets for autocompletes? I think taggit probably custom autocomplete views, so these lines are unneeded.